src/components/projectStatus/ProjectStatusHome.tsx

```diff
@@ -25,21 +26,37 @@ const ProjectStatusHome: React.FC = () => {
       },
     });
 
-  if (projectsLoading) {
+  const [{ loading: tableGroupsLoading, data: tableGroupsData, error: tableGroupsError }] =
+    useAxios({
+      method: "GET",
+      url: apiUrl(Service.EXPO, "/tablegroups"),
+      params: {
+        hexathon: currentHexathon.id,
+      },
+    });
+
+  if (projectsLoading || tableGroupsLoading) {
     return <LoadingDisplay />;
```

Comment:
So this other GET request is not needed. You are making another large request just to filter out the table group name for each project. Instead of doing that, what you can do is get the table group name from the project. If it doesn't exist on the existing request, what you can do is update the [Prisma include](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries) field on the backend api request to include the data you need. This will significantly simplify your code and your frontend changes should only be a few lines total.

Reply:
Sounds good, I can make a PR to api first then. Also, this could change how we retrieve table group names for the project gallery page, too. Like instead of needing to make a separate request like the one you highlighted, maybe it can just be one request to get all the necessary info (referring to this specific [file](https://github.com/HackGT/timber/blob/bce2f697ac5773bc801447eaf078ad3fa8452912/src/components/projectGallery/ProjectGallery.tsx#L54C47-L54C62)).

Reply:
nice callout @ayush-goyal, also yeah @nicolasC777 if you could update that as well, it'd be great, should basically be the same code

Reply:
Yeah good catch @nicolas-cai , that page should be updated as well. I had caught a similar issue and fixed it https://github.com/HackGT/timber/commit/e178707975bf3081884c4ce0af7a0684ca06676f, but the project gallery should also be updated.